File: src/main/ipc/engine.ts
import { IPC } from '@shared/protocol/ipc';
import { ipcMain } from 'electron';
import type { IpcDeps } from './deps';

export function registerEngineHandlers(deps: IpcDeps): void {
  ipcMain.handle(IPC.engineStatus, () => deps.engineHost.status());
}

File: shell/internal/bridge/engine.go
package bridge

import "github.com/kirathecat/kira-studio/shell/internal/appcore"

// EngineStatus mirrors src/shared/protocol/ipc.ts's EngineStatus.
type EngineStatus struct {
	Alive bool `json:"alive"`
	PID   *int `json:"pid"`
}

type EngineService struct {
	Deps appcore.Deps
}

func (s *EngineService) Status() (EngineStatus, error) {
	if s.Deps.EngineHost == nil || !s.Deps.EngineHost.Alive() {
		return EngineStatus{Alive: false, PID: nil}, nil
	}
	pid := s.Deps.EngineHost.PID()
	return EngineStatus{Alive: true, PID: &pid}, nil
}
